core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1312,14 +1320,16 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
       if (!isActive) return
       val partitionReplicaAssignment = zkClient.getReplicaAssignmentForTopics(immutable.Set(topic))
       val partitionsToBeAdded = partitionReplicaAssignment.filter(p =>
-        !controllerContext.partitionReplicaAssignment.contains(p._1))
+        controllerContext.partitionReplicaAssignment(p._1).isEmpty)
```

Comment:
Perhaps we can change p to case (topicPartion, _) and use topicPartition instead of p._1?

Reply:
Yes, that's better.